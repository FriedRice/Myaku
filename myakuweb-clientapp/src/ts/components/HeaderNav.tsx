/** @module Nav component for the main MyakuWeb header */

import React from 'react';
import myakuLogoUrl from '../../images/myaku-logo.svg';

const MYAKU_GITHUB_LINK = 'https://github.com/FriedRice/Myaku';

const HeaderNav: React.FC<{}> = function() {
    return (
        <nav className='header-nav'>
            <a className='nav-logo' href='/'>
                <img
                    className='myaku-logo'
                    src={myakuLogoUrl}
                    alt='Myaku logo'
                />
            </a>
            <ul className='nav-link-list'>
                <li>
                    <a aria-label='Myaku GitHub' href={MYAKU_GITHUB_LINK}>
                        Github
                    </a>
                </li>
            </ul>
        </nav>
    );
};

export default HeaderNav;
