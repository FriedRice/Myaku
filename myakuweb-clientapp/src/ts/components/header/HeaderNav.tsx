/** @module Nav component for the main MyakuWeb header */

import React from 'react';
import myakuLogoUrl from 'images/myaku-logo.svg';

interface HeaderNavProps {
    onReturnToStart: () => void;
}
type Props = HeaderNavProps;

const MYAKU_GITHUB_LINK = 'https://github.com/FriedRice/Myaku';


const HeaderNav: React.FC<Props> = function(props) {
    function handleReturnToStart(event: React.SyntheticEvent): void {
        event.preventDefault();
        props.onReturnToStart();
    }

    return (
        <nav className='header-nav'>
            <a className='nav-logo' href='/' onClick={handleReturnToStart}>
                <img
                    className='myaku-logo'
                    src={myakuLogoUrl}
                    alt='Myaku logo'
                />
            </a>
            <ul className='nav-link-list'>
                <li>
                    <a aria-label='Myaku GitHub' href={MYAKU_GITHUB_LINK}>
                        Github
                    </a>
                </li>
            </ul>
        </nav>
    );
};

export default HeaderNav;
