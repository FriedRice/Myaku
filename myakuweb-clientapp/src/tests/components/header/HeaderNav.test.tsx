/**
 * Tests for the HeaderNav component.
 * @module tests/components/header/HeaderNav.test
 */

import HeaderNav from 'ts/components/header/HeaderNav';
import React from 'react';
import { shallow } from 'enzyme';


describe('<HeaderNav />', function() {
    it('renders correctly', function() {
        const wrapper = shallow(<HeaderNav />);
        expect(wrapper).toMatchSnapshot();
    });
});
