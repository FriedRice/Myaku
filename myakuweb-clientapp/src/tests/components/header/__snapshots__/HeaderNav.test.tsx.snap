// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderNav /> renders correctly 1`] = `
<nav
  className="header-nav"
>
  <Link
    className="nav-logo"
    to="/"
  >
    <img
      alt="Myaku logo"
      className="myaku-logo"
      src="test-file-stub"
    />
  </Link>
  <ul
    className="nav-link-list"
  >
    <li>
      <a
        aria-label="Myaku GitHub"
        href="https://github.com/FriedRice/Myaku"
      >
        Github
      </a>
    </li>
  </ul>
</nav>
`;
